### PR TITLE
Add doxygen job and GH pages

### PR DESCRIPTION
**Pull request by Ryanf55**
# Purpose

Create and host API documentation for grid_map_geo. Since this library is largely intended to be used in other projects, it is important to have a well documented C++ API. 

Luckliy, we can use Doxygen to generate the docs, and view them on github pages with a few actions.

# Demo

https://ryanf55.github.io/grid_map_geo/index.html

# Reference

https://github.com/marketplace/actions/doxygen-action

### CODE REVIEW

**Comment by Ryanf55** [comment]
In order for the build to pass, the repo admin must enable Github pages. 
https://github.com/ethz-asl/grid_map_geo/actions/runs/7552085852/job/20560362797?pr=49#step:5:162

See this screenshot for the appropriate settings:
image

If you don't want to do the pages stuff quite yet, but want the docs, I can split this PR in two.

File: .github/workflows/doxygen.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Build Test
```
Comment (Jaeyoung-Lim) [inline .github/workflows/doxygen.yml:1]: Maybe we should call it what it actually is :)

**Review by Jaeyoung-Lim — CHANGES_REQUESTED** [review]
This is amazing!

No other comment than setting the name of the action to reflect what is actually inside

**Comment by Jaeyoung-Lim** [comment]
Merging to see if it works :)

**Comment by Ryanf55** [comment]
> Merging to see if it works :)

Sort of? 
https://ethz-asl.github.io/grid_map_geo/

Hahah, it's the README, but not API docs. I'll look at it tonight.

**Pull request merged**